datafusion/core/src/datasource/physical_plan/file_scan_config.rs

```diff
@@ -527,6 +526,38 @@ mod tests {
         assert_eq!(col_indices, None);
     }
 
+    #[test]
+    fn physical_plan_config_no_projection_tab_cols_as_field() {
+        let file_schema = aggr_test_schema();
+
+        // make a table_partition_col as a field
+        let table_partition_col = Field::new(
+            "date".to_owned(),
```

Comment:
You don't have to call `to_owned()` here -- `Field::new` will do it for you

```suggestion
            "date",
```